Added new changes in App

TobiA34 opened the pull request:
- Changed stack view in salaryInfoView
- Created a Accordion view
- Created a Salary view
- Created a Student Loan view
- Created a Submit view
- Created a Tax view
- Created file and folder for InputManager
- Removed comments in NICalculator
- Updated calculate function in student loan calculator
- Removed comments in TaxCalculator
-  Created FormInputDelegate
- Created a validation Manager
- Removed code in Salary break down view controller and replaced it with accordion view
- Updated a code in salary form view controller
- Updated a code in salary form view model
- Updated unit and ui test

TobiA34 committed:
- Add app icon
- Changed stack view in salaryInfoView
- Created a Accordion view
- Created a Salary view
- Created a Student Loan view
- Created a Submit view
- Created a Tax view
- Created file and folder for InputManager
- Removed comments in NICalculator
- Updated calculate function in student loan calculator
- Removed comments in TaxCalculator
-  Created FormInputDelegate
- Created a validation Manager
- Removed code in Salary break down view controller and replaced it with accordion view
- Updated a code in salary form view controller
- Updated a code in salary form view model
- Updated unit and ui test

tunds commented on Salary App/Base/Views/SalaryBreakdown/AccordionView.swift:
```diff
@@ -0,0 +1,303 @@
+    
+    private lazy var dailyStackView: SalaryHorizontalStackView = {
+        let stackView = SalaryHorizontalStackView()
+        stackView.translatesAutoresizingMaskIntoConstraints = false
```
You don't need to set this here all this configuration should be done in the setup function within the component, this is the same for the other views. This should just be 

```

private lazy var dailyStackView: SalaryHorizontalStackView = {
        let vw = SalaryHorizontalStackView()
        return vw
}

```

All styling should be done inside of the component, things like actions you can use delegates or closures for communication

tunds commented on Salary App/Base/Views/SalaryBreakdown/AccordionView.swift:
```diff
@@ -0,0 +1,303 @@
+        }
+    }
+    
+    @objc func touchHappen() {
```
For actions you normally want to use something like `didTouchToggleIcon`

tunds commented on Salary App/Base/Views/SalaryForm/PensionView.swift:
```diff
@@ -0,0 +1,101 @@
+ 
+class PensionView: UIView {
+    
+    weak var delegate: FormInputDelegate!
```
Always make delegates optional, don't force unwrap them.

tunds commented on Salary App/Base/Views/SalaryForm/PensionView.swift:
```diff
@@ -0,0 +1,101 @@
+    
+   @objc func enterPension() {
+       if let pensionInput = pensionTextField.text,
+          let pensionDelegate = delegate {
```
No need to make this part of the optional wrapping, you can use an optional property like

```
if let pensionInput = pensionTextField.text {
  delegate?.didEnter(id: id, input: pensionInput)
}

```

tunds commented on Salary App/Base/Views/SalaryForm/SalaryView.swift:
```diff
@@ -0,0 +1,110 @@
+    
+    @objc func enterSalary() {
+         if let salaryInput = salaryTextField.text,
+            let salaryDelegate = delegate {
```
Same here as above

tunds commented on Salary App/Base/Views/SalaryForm/SalaryView.swift:
```diff
@@ -0,0 +1,110 @@
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
```
Clean up any dead space

tunds commented on Salary App/Base/Views/SalaryForm/StudentLoanView.swift:
```diff
@@ -0,0 +1,137 @@
+                StudentLoanPlan.postgraduateLoan.rawValue
+    ]
+    
+    private let salaryTypeTextField = UITextField()
```
With the picker why do we not create it like we create the other views above?

tunds commented on Salary App/Base/Views/SalaryForm/StudentLoanView.swift:
```diff
@@ -0,0 +1,137 @@
+        return taxCodeLbl
+    }()
+    
+    let pickerView = UIPickerView()
```
With the picker why do we not create it like we create the other views above?

tunds commented on Salary App/Base/Views/SalaryForm/SubmitView.swift:
```diff
@@ -0,0 +1,59 @@
+
+class SubmitView: UIView {
+    
+    weak var delegate: ButtonViewDelegate!
```
Again all delegates, should be optional

TobiA34 committed:
- Removed translatesAutoresizingMaskIntoConstraints on computed properties in accordion view
- changed function name from touch happen to didTouchToggleIcon in accordion view
- Changed delegate to be optional in PensionView
- Changed delegate to be optional in SalaryView
- Changed delegate to be optional in StudentLoanView
- Changed delegate to be optional in SubmitView
- Cleaned up validation manager code
- Cleaned up salary form view controller code
- Cleaned up code in SalaryFormViewModel
- Cleaned up code in apptest and uitests
